pkg/artifactory/resource_artifactory_backup.go

```diff
@@ -0,0 +1,251 @@
+			Description: `(Optional) Flag to enable or disable the backup config. Default value is "true".`,
+		},
+		"cron_exp": {
+			Type:             schema.TypeString,
+			Required:         true,
+			ValidateDiagFunc: validation.ToDiagFunc(validateCron),
+			Description:      `(Required) Cron expression to control the backup frequency.`,
+		},
+		"retention_period_hours": {
+			Type:             schema.TypeInt,
+			Optional:         true,
+			Default:          168,
+			ValidateDiagFunc: validation.ToDiagFunc(validation.IntAtLeast(0)),
+			Description:      `(Optional) The number of hours to keep a backup before Artifactory will clean it up to free up disk space. Applicable only to non-incremental backups. Default value is 168 hours ie: 7 days.`,
+		},
+		"excluded_repositories": {
+			Type:        schema.TypeList,
+			Optional:    true,
+			Elem:        &schema.Schema{Type: schema.TypeString},
+			Description: `(Optional) list of excluded repositories from the backup. Default is empty list.`,
+		},
+		"create_archive": {
+			Type:        schema.TypeBool,
+			Optional:    true,
+			Default:     false,
+			Description: `(Optional) If set to true, backups will be created within a Zip archive (Slow and CPU intensive). Default value is 'false'`,
+		},
+		"exclude_new_repositories": {
+			Type:        schema.TypeBool,
+			Optional:    true,
+			Default:     false,
+			Description: `(Optional) When set to true, new repositories will not be automatically added to the backup. Default value is 'false'.`,
+		},
+		"send_mail_on_error": {
+			Type:        schema.TypeBool,
+			Optional:    true,
+			Default:     true,
+			Description: `(Optional) If set to true, all Artifactory administrators will be notified by email if any problem is encountered during backup. Default value is 'true'.`,
+		},
+	}
+	var resourceBackupRead = func(_ context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
+		backups := &Backups{}
+		backup := unpackBackup(d)
+
+		_, err := m.(*resty.Client).R().SetResult(&backups).Get("artifactory/api/system/configuration")
+		if err != nil {
+			return diag.Errorf("failed to retrieve data from API: /artifactory/api/system/configuration during Read")
+		}
+
+		matchedBackup := Backup{}
+		for _, iterBackup := range backups.BackupArr {
+			if iterBackup.Key == backup.Key {
+				matchedBackup.Key = iterBackup.Key
+				matchedBackup.Enabled = iterBackup.Enabled
+				matchedBackup.CronExp = iterBackup.CronExp
+				matchedBackup.ExcludeNewRepositories = iterBackup.ExcludeNewRepositories
+				matchedBackup.CreateArchive = iterBackup.CreateArchive
+				matchedBackup.RetentionPeriodHours = iterBackup.RetentionPeriodHours
+				matchedBackup.SendMailOnError = iterBackup.SendMailOnError
+				for _, repo := range iterBackup.ExcludedRepositories.RepositoryRef {
+					matchedBackup.ExcludedRepositories = append(matchedBackup.ExcludedRepositories, repo)
+				}
+				break
+			}
+		}
+		var backupClass = ignoreHclPredicate("class", "rclass")
```

Comment:
I am pretty certain this is already defined in the same place where `ignoreHclPredicate` is defined